tests/templates/test_integration.py

```diff
@@ -176,558 +138,347 @@
-        diffable2 = list(diffable2.values())
-
-        # Combine diffable inputs to allow passing with *
-        diffable = diffable1 + diffable2
+        keys_diffable = [*diffable]
+        diffable = list(diffable.values())
 
         # Turn into correct format
         diffable = [to_var(i) for i in diffable]
 
         # Generate qnode
         dev = qml.device('default.qubit', wires=2)
-        circuit = qnode_qubit_args(dev, interface, template1, template2, len(diffable1))
+
+        @qml.qnode(dev, interface=interface)
+        def circuit(*diffable, keys_diffable=None, nondiffable=None):
+            # Turn diffables back into dictionary
+            dict = {key: item for key, item in zip(keys_diffable, diffable)}
+
+            # Merge with nondiffables
+            dict.update(nondiffable)
+
+            # Add number of wires
+            dict['wires'] = range(2)
+
+            template(**dict)
+            return qml.expval(qml.Identity(0))
 
         # Check that execution does not throw error
-        circuit(*diffable, keys_diffable1=keys_diffable1, keys_diffable2=keys_diffable2,
-                nondiffable1=nondiffable1, nondiffable2=nondiffable2)
+        circuit(*diffable, keys_diffable=keys_diffable, nondiffable=nondiffable)
 
-    @pytest.mark.parametrize("template1, diffable1, nondiffable1", CV_DIFFABLE_NONDIFFABLE)
-    @pytest.mark.parametrize("template2, diffable2, nondiffable2", CV_DIFFABLE_NONDIFFABLE)
+    @pytest.mark.parametrize("template, diffable, nondiffable", CV_DIFFABLE_NONDIFFABLE)
     @pytest.mark.parametrize("interface, to_var", INTERFACES)
-    def test_integration_cv_diffable(self, template1, diffable1, nondiffable1,
-                                     template2, diffable2, nondiffable2,
-                                     interface, to_var, gaussian_device_2_wires):
+    def test_cv_qnode_primary_args(self, template, diffable, nondiffable,
+                                   interface, to_var, gaussian_device_2_wires):
         """Tests integration of cv templates passing differentiable arguments as positional arguments to qnode."""
 
         # Extract keys and items
-        keys_diffable1 = [*diffable1]
-        diffable1 = list(diffable1.values())
-        keys_diffable2 = [*diffable2]
-        diffable2 = list(diffable2.values())
-
-        # Combine diffable inputs to allow passing with *
-        diffable = diffable1 + diffable2
+        keys_diffable = [*diffable]
+        diffable = list(diffable.values())
 
         # Turn into correct format
         diffable = [to_var(i) for i in diffable]
 
         # Generate qnode
-        circuit = qnode_cv_args(gaussian_device_2_wires, interface, template1, template2, len(diffable1))
+        @qml.qnode(gaussian_device_2_wires, interface=interface)
+        def circuit(*diffable, keys_diffable=None, nondiffable=None):
+            # Turn diffables back into dictionary
+            dict = {key: item for key, item in zip(keys_diffable, diffable)}
```

Comment:
(Minor suggestion): this also seems to work
```suggestion
            dict = dict(zip(keys_diffable, diffable))
```